src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -107,6 +101,14 @@ object BitPat {
     apply("b" + x.litValue.toString(2).reverse.padTo(len, "0").reverse.mkString)
   }
 
+    /**
+    * Allows [[ChiselEnum]] to be used where a BitPat is expected.
+    */
+  def apply(x: EnumType): BitPat = {
+    val len = if (x.isWidthKnown) x.getWidth else 0
+    apply("b" + x.litValue.toString(2).reverse.padTo(len, "0").reverse.mkString)
```

Comment:
You could still re-use the apply function, i.e., do something like:
```
def apply(x: EnumType): BitPat = apply(x.litValue.U(n.getWidth).W)
```

Reply:
👍🏻